docs/standards/python_coding_standards.md

```diff
@@ -0,0 +1,304 @@
+# Python Standards
+
+This document defines the internal development standards for building backend services in Python ([FastAPI](https://fastapi.tiangolo.com/tutorial/)) within Defra. It adheres strictly to the [UK Government Digital Service (GDS) Python style guide](https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/python/python.html) and [PEP 8](https://peps.python.org/pep-0008/), ensuring code is clear, consistent, and maintainable across all AI services (using tools like [LangChain](https://python.langchain.com/docs/introduction/), [LangGraph](https://langchain-ai.github.io/langgraph/tutorials/introduction/), etc.). All engineers must follow these conventions when writing Python code for backend services.
+
+**Note:** Python should **ONLY** be used for creating backend services related to AI or data.
+
+**Note:** Python should **NOT** be used for frontend development. The GDS Service Manual has information on using client-side JavaScript. For server-side JavaScript we use Node.js. The principles of [Progressive Enhancement](https://www.gov.uk/service-manual/technology/using-progressive-enhancement) should be used for the design and build of frontend digital services.
+
+## Function Annotations and Typing Conventions
+- Use Type Hints for All Functions: All function and method definitions should include Python type annotations for parameters and return values (PEP 484 style). This improves readability and helps static analysis.
```

Comment:
Is there a reason we should include them by default?  We wouldn't say that JSDoc should be used for all JavaScript as it may be an unnecessary overhead so why would we be inconsistent with Python?

Is there a data science-y/AI-y reason?

Reply:
Heh - have you tried _reading_ Python code?! 🤣